keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,742 @@
 
-## Sharing COSI Created Buckets
-This is the greenfield -> brownfield access use case, where COSI has created the `Bucket` instance and the driver has provisioned a new bucket. Now, we want to share access to this bucket in other namespaces.
+More information about BucketAccess is [here](#bucketaccess)
 
-![ShareBucket Workflow](images/share-bucket.png)
+```
+    Bucket Access - bar-$uuid
+    |--------------------------------------|
+    | name: bar-$uuid                      |
+    | spec:                                |
+    |   bucketAccessClassName: bac1        |
+    |   parameters:                        |
+    |     key: value                       |
+    |   bucket:                            |
+    |     name: br-$uuid                   |
+    |     uuid: $uuid                      | 
+    | status:                              |
+    |   credentialsSecret: nil             |
+    |   conditions:                        |
+    |   - type: AccessGranted              |
+    |     status: False                    |
+    |--------------------------------------|
+```
 
-If the bucket sharing is all within the same namespace then each `BucketAccessRequest` (in the namespace) only needs to reference the existing BR.
+###### 3. COSI calls driver to create BucketAccess
 
-Here is the workflow:
+The COSI sidecar, which runs alongside each of the drivers, listens for BucketAccess events. All but the sidecar with the specified provisioner will ignore the BucketAccess object. Only the appropriate sidecar will make a gRPC request to its driver to generate credentials for a Bucket in the OSP.
 
-+ A `Bucket` instance created by COSI must be cloned by the admin (each `Bucket` instance needs a unique name).
-+ The user creates a `BucketRequest` (BR) that specifies the `Bucket` instance name (BR.bucketInstanceName) provided by the admin.
-+ The user creates a `BucketAccessRequest` (BAR) - see [grant bucket access](#grant-bucket-access) -- and the BAR references their BR.
+More information about COSI gRPC API is [here](#cosi-driver-grpc-api)
 
-## Delete Bucket
+```
+    COSI Driver (s3.amazonaws.com)
+    |------------------------------------------|
+    | grpc ProvisionerGrantBucketAccess({      |
+    |      "name": "bar-$uuid",                |
+    |      "bucketID": "br-$uuid",             |
+    |      "parameters": {                     |
+    |         "key": "value"                   |
+    |      }                                   |
+    | })                                       |
+    |------------------------------------------|
+```
+Each BucketAccess is meant to map to one account (or its equivalent) in the OSP with privileges to access the bucket. Once the credentials are successfully generated in the OSP, a secret by the name specified in the BucketAccessRequest is created and the credential values are stored in it. The secret resides in the namespace of the BucketAccessRequest. 
```

Comment:
I recall talking about how service-account oriented access control would work before, but I don't see anything here.  I think it's imperative to design for it.

Specifically, I think you need a way to specify which SA to grant access to and a way to denote "access was granted" that isn't a long-lived token (and might not be a token at all).  Now you have to abstract that so the same request works for both modes (or your user is coupled to one implementation).

Reply:
Strongly agree. I've spelled out the reasons for going the access/secret token route for alpha in other comments.